frontend-react/src/hooks/network/ManagePublicKey/ManagePublicKeyHooks.ts

```diff
@@ -0,0 +1,183 @@
+export interface RequestCompletePayload {
+    response: {
+        id?: string;
+        timestamp?: string;
+    };
+}
+
+interface FileSelectedPayload {
+    file: File;
+}
+
+type ManagePublicKeyActionPayload =
+    | RequestCompletePayload
+    | FileSelectedPayload
+    | null;
+
+interface ManagePublicKeyAction {
+    type: ManagePublicKeyActionType;
+    payload?: ManagePublicKeyActionPayload; // reset actions will have no payload
+}
+
+type ManagePublicKeyReducer = (
+    state: ManagePublicKeyState,
+    action: ManagePublicKeyAction
+) => ManagePublicKeyState;
+
+export const INITIAL_STATE = {
+    fileInputResetValue: 0,
+    fileContent: "",
+    fileName: "",
+    selectedSender: "",
+    reportId: "",
+    successTimestamp: "",
+    localError: "",
+};
+
+function getInitialState(): ManagePublicKeyState {
+    return INITIAL_STATE;
+}
+
+// state for resetting / setting state at beginning of submission
+function getPreSubmitState(): Partial<ManagePublicKeyState> {
+    return {
+        ...pick(INITIAL_STATE, ["reportId", "successTimestamp", "localError"]),
+    };
+}
+
+// update state when file is selected in form
+function calculateFileSelectedState(
+    state: ManagePublicKeyState,
+    payload: FileSelectedPayload
+): Partial<ManagePublicKeyState> {
+    const { file } = payload;
+    try {
+        // look at the filename extension.
+        const fileNameArray = file.name.split(".");
+        const uploadFileExtension = fileNameArray[fileNameArray.length - 1];
+
+        if (uploadFileExtension.toUpperCase() !== Format.PEM) {
+            return {
+                ...state,
+                localError: "The file type must be of type .pem",
+            };
+        }
+
+        if (file.size > PAYLOAD_MAX_BYTES) {
+            return {
+                ...state,
+                localError: `The file '${file.name}' is too large. The maximum file size is ${PAYLOAD_MAX_KBYTES}k`,
+            };
+        }
+
+        // previously loading file contents here
+        // since this is an async action we'll do this in the calling component
+        // prior to dispatching into the reducer, and handle the file content in local state
+        const contentType = ContentType.PEM;
+        const fileType = Format.PEM;
+
+        return {
+            ...state,
+            fileType,
+            fileName: file.name,
+            contentType,
+        };
+    } catch (err: any) {
+        console.warn(err);
+        return {
+            ...state,
+            localError: `An unexpected error happened: '${err.toString()}'`,
+        };
+    }
+}
+
+// update state when API request is complete
+function calculateRequestCompleteState(
+    state: ManagePublicKeyState,
+    payload: RequestCompletePayload
+): Partial<ManagePublicKeyState> {
+    const {
+        response: { id, timestamp },
+    } = payload;
+
+    return {
+        fileInputResetValue: state.fileInputResetValue + 1,
```

Comment:
I understand the reasoning for this in the existing FileHandler, but my feeling is that it's an anti-pattern since we shouldn't have to force a re-rendering by keeping track of a separate key.  If anything, it should just be a derived value from the data we'd actually want to submit.